server/src/main/java/io/crate/execution/jobs/CumulativePageBucketReceiver.java

```diff
@@ -57,6 +60,12 @@ public class CumulativePageBucketReceiver implements PageBucketReceiver {
 
     private static final Logger LOGGER = LogManager.getLogger(CumulativePageBucketReceiver.class);
 
+    /**
+     * Completing queries has higher priority than starting queries
+     * See {@link CollectTask}
+     **/
+    private static final Priority PRIORITY = Priority.HIGH;
```

Comment:
Isn't the priority here related to the query type as well? Afaik this is also used for some queries against system tables, e.g. joins.

Reply:
query type info is unfortunately not available in the merge phase.

I changed the comment a bit in a fixup, to clarify the intention